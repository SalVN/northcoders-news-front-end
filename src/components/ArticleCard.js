import React from 'react';
import PropTypes from 'prop-types';
import { BrowserRouter as Router, Link } from 'react-router-dom';

const ArticleCard = function (props) {
  return (
    <div className='box'>
      <article className='media'>
        <div className='media-left'>
          {props.number + 1}
        </div>
        <div className='media-left'>
          <button className='button' onClick={props.voteHandler.bind(this, 'up', props.id)}><i className="fa fa-arrow-up" aria-hidden="true"></i></button>
          <div>{props.votes}</div>
          <button className='button' onClick={props.voteHandler.bind(this, 'down', props.id)}><i className="fa fa-arrow-down" aria-hidden="true"></i></button>
        </div>
        <div className='media-content'>
          <div className='content'>
            <Link to={`/articles/${props.id}`}>
              <h3 className='title is-3'>{props.title}</h3>
            </Link>
            <p>
              <span>added by: {props.author}</span>
              <span>tags: {props.tags}</span>
              <span>comments: {props.comment_count}</span>
            </p>
          </div>
        </div>
      </article>
    </div>
  );
};

ArticleCard.propTypes = {
  title: PropTypes.string.isRequired,
  votes: PropTypes.number.isRequired,
  number: PropTypes.number.isRequired,
  author: PropTypes.string.isRequired,
  id: PropTypes.string.isRequired,
  tags: PropTypes.string.isRequired,
  comment_count: PropTypes.number.isRequired,
  voteHandler: PropTypes.func.isRequired
};


export default ArticleCard;
