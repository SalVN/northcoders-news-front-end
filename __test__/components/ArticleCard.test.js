import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';

import ArticleCard from '../../src/components/ArticleCard';

describe('ArticleCard', () => {
    const article = {
        _id: '59b01acf006c8dbca914672f',
        title: 'Football is fun',
        body: 'something',
        belongs_to: 'football',
        __v: 0,
        votes: 3,
        created_by: 'northcoder',
        comment_count: 0
    };
    it('is a function', () => {
        expect(typeof ArticleCard).toEqual('function');
    });

    it('renders', () => {
        const enzymeWrapper = shallow(<ArticleCard
            title={article.title}
            votes={article.votes}
            number={1}
            author={article.created_by}
            id={article._id}
            tags={article.belongs_to}
            comment_count={article.comment_count}
            voteHandler={x => x}
        />);
        expect(enzymeWrapper.children().length).toEqual(1);
    });

    it('renders correctly', () => {
        const tree = renderer.create(
            <MemoryRouter>
                <ArticleCard
                    title={article.title}
                    votes={article.votes}
                    number={1}
                    author={article.created_by}
                    id={article._id}
                    tags={article.belongs_to}
                    comment_count={article.comment_count}
                    voteHandler={x => x}
                />
            </MemoryRouter>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
});