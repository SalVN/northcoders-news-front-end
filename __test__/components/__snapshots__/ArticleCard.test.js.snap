// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleCard renders correctly 1`] = `
<div
  className="box article-box-style"
>
  <article
    className="media"
  >
    <div
      className="media-left article-number is-hidden-mobile"
    >
      2
    </div>
    <div
      className="media-left is-hidden-mobile"
    >
      <button
        className="button is-black is-inverted is-small"
        onClick={[Function]}
      >
        <i
          aria-hidden="true"
          className="fa fa-arrow-up"
        />
      </button>
      <p
        className="article-card-votes"
      >
        <strong>
          3
        </strong>
      </p>
      <button
        className="button is-black is-inverted is-small"
        onClick={[Function]}
      >
        <i
          aria-hidden="true"
          className="fa fa-arrow-down"
        />
      </button>
    </div>
    <div
      className="media-content"
    >
      <div
        className="content"
      >
        <a
          href="/articles/59b01acf006c8dbca914672f"
          onClick={[Function]}
        >
          <h3
            className="title is-4 article-card-title"
          >
            <strong>
              Football is fun
            </strong>
          </h3>
        </a>
        <div
          className="columns is-gapless is-desktop"
        >
          <div
            className="column is-narrow"
          >
            <span
              className="article-card-subheading-key"
            >
              added by:
            </span>
          </div>
          <div
            className="column is-narrow"
          >
            <span>
              <span
                className="article-card-subheading-key"
              >
                tags:
              </span>
              <a
                href="/topics/football/articles"
                onClick={[Function]}
              >
                <span
                  className="article-card-subheading-value"
                >
                  <strong>
                    football
                  </strong>
                </span>
              </a>
            </span>
            <span
              className="article-card-subheading-key"
            >
              comments:
            </span>
            <a
              href="/articles/59b01acf006c8dbca914672f#comments"
            >
              <span
                className="article-card-subheading-value"
              >
                <strong>
                  0
                </strong>
              </span>
            </a>
          </div>
        </div>
      </div>
    </div>
  </article>
</div>
`;
